### ISSUE TITLE
flocker to support eucalyptus ebs

### ISSUE THREAD

**Issue by gufranmmu** [ClusterHQ/flocker#2797]
By default aws sdk connects to amazon endpoint. Add functionality in flocker ebs.py so that user can provide endpoint url. Example:
```
    ec2_resource = connection.resource("ec2", region_name=region, endpoint_url="http://compute.esclor86.eecloud.privatecloud.net:8773")
```

**Comment by wallnerryan** [comment]
Hi @gufranmmu. I am unfarmiliar with eucalyptus. According to https://github.com/eucalyptus/eucalyptus/wiki is implements the ec2 API for storage? E.g. by pointing it at something like ` endpoint_url="http://compute.esclor86.eecloud.privatecloud.net:8773` your hope it to have a eucalyptus compatible flocker cluster point at and ebs implementation of a eucalyptus private cloud? (sorry if I got some of the terminology wrong)

**Comment by gufranmmu** [comment]
yes, that's exactly what it mean is to be able to provide endpoint URL for EBS via the agent.yml config file :) Thank you for replying!

https://support.eucalyptus.com/hc/en-us/articles/200844844-Usage-of-AWS-CLI-with-Eucalyptus

**Comment by wallnerryan** [comment]
No problem. I will let our engineering team know of your request.
